### PR TITLE
Move FCM class to organisms, convert rendering of it to include_block

### PR DESCRIPTION
**Pull request by Scotchester**
This PR addresses a few things, most urgently the inability to compare revisions on any page that has an FCM (due to the outdated `FeaturedContent` `template` property). 

It moves the Python code to organisms to be consistent with that change in the front-end code, and removes the `macro`-based rendering setup to the more standard Wagtail `include_block` method. 

It also corrects a lingering template issue with empty link list markup when no link was set up.

## Changes

- Moves the `RelatedPosts` class from `molecules.py` to `organisms.py`
- Updates page templates to render `RelatedPosts` blocks via `include_block` instead of `macro.render()`
- Fixes `related-posts.html` template logic to not render empty link list markup if no links are present

## Removals

- Unused `macro_link` stuff in `related-posts.html`

## Testing

1. Pull branch
2. Visit pages of each type with an FCM and confirm that they are still working
   - Browse: http://localhost:8000/es/prepararse-para-comprar-casa/hipotecas/
   - Sublanding: http://localhost:8000/consumer-tools/auto-loans/
   - Sublanding Filterable: http://localhost:8000/consumer-tools/auto-loans/
   - Browse Filterable: One doesn't exist that uses an FCM. You can make your own.
   - Newsroom Landing: Doesn't currently uses an FCM. You can make your own.

## Checklist

- [ ] PR has an informative and human-readable title
- [ ] Changes are limited to a single goal (no scope creep)
- [ ] Code can be automatically merged (no conflicts)
- [ ] Code follows the standards laid out in the [development playbook](https://github.com/cfpb/development)
- [ ] Passes all existing automated tests
- [ ] Any _change_ in functionality is tested
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged / future todos are captured in comments
- [ ] Visually tested in supported browsers and devices (see checklist below :point_down:)
- [ ] Project documentation has been updated
- [ ] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

### CODE REVIEW

File: cfgov/regulations3k/jinja2/regulations3k/browse-regulation.html [hunk 1]
```diff
@@ -1,7 +1,6 @@
 {% extends 'regulations3k/regulations3k-side-nav.html' %}
 
 {% import 'form_block.html' as form_block with context %}
-{% import 'organisms/featured-content.html' as featured_content with context %}
```
Comment (chosak) [inline cfgov/regulations3k/jinja2/regulations3k/browse-regulation.html]: Am I reading this correctly that this organism isn't actually used on this Regs3k page type? It doesn't seem to be one of the [supported block types](https://github.com/cfpb/cfgov-refresh/blob/master/cfgov/regulations3k/models/pages.py#L201)?

Comment (Scotchester) [reply cfgov/regulations3k/jinja2/regulations3k/browse-regulation.html]: That does seem correct to my eyes, as well. @willbarton or @contolini, can you confirm?

Comment (contolini) [reply cfgov/regulations3k/jinja2/regulations3k/browse-regulation.html]: Hmmm I bet we copied a template when we started regs3k and accidentally left some extraneous stuff.

**Comment by chosak** [comment]
I don't want to blow up the scope of your PR, but while you're in here:

- Is it a bug that [`category`](https://github.com/cfpb/cfgov-refresh/pull/4699/files#diff-dcf2a8703e945442ba8591d07085a645R939) is not required, but a null category [is not checked for in the template](https://github.com/cfpb/cfgov-refresh/pull/4699/files#diff-1960b1d27b4bfc1a4eb279c986d73522R64)? If you leave the category blank, the page validation passes, but you see a None on the page:

    image
- Similarly, [`image`](https://github.com/cfpb/cfgov-refresh/pull/4699/files#diff-dcf2a8703e945442ba8591d07085a645R946) is also optional. I don't think that [this](https://github.com/cfpb/cfgov-refresh/pull/4699/files#diff-dcf2a8703e945442ba8591d07085a645R946) `required=False` is doing what we want, because [this](https://github.com/cfpb/cfgov-refresh/pull/4699/files#diff-1960b1d27b4bfc1a4eb279c986d73522R96) `elif value.image` is still evaluating as true even if you don't specify an image. This is probably because the `ImageBasic` block does always exist; what we really want to check is `value.image.upload`. See also https://github.com/wagtail/wagtail/issues/2665 which explains that `required` doesn't have any affect on a sub-StructBlock. If you don't specify an image, you get this kind of thing in the page source:

    image

    Per this logic, [this](https://github.com/cfpb/cfgov-refresh/pull/4699/files#diff-1960b1d27b4bfc1a4eb279c986d73522R91) `if value.video` is also probably always true, although here we're correctly checking against `.url` and `.id`.
- Conversely, our `Hyperlink` [does seem to have](https://github.com/cfpb/cfgov-refresh/blob/master/cfgov/v1/atomic_elements/atoms.py#L77) some kind of custom `required` logic. We don't enforce that links have URLs, which allows you to enter links that don't point anywhere:

    image

    Validating this would require a change to `Hyperlink`, but the template for the FCM could add a check for `value.links[0].url` here to at least skip empty links.
- `post_link_text` [is included in the template](https://github.com/cfpb/cfgov-refresh/pull/4699/files#diff-1960b1d27b4bfc1a4eb279c986d73522R74) even though [it is also not required](https://github.com/cfpb/cfgov-refresh/pull/4699/files#diff-dcf2a8703e945442ba8591d07085a645R944). Should it default to the page title? If you don't enter a title, a weird empty link is inserted in the page source:

    image

**Comment by Scotchester** [comment]
@chosak All good catches, thanks! Fixed them up in the latest commit, then rebased on `master` and force-pushed.

**Pull request merged**